src/igraph/io/objects.py

```diff
@@ -645,29 +645,29 @@ def _export_graph_to_list_dict(
 
     This function is the reverse of Graph.ListDict.
 
-    @param use_vids (bool): whether to label vertices in the output data
+    Example:
+
+      >>> g = Graph.Full(3)
+      >>> g.to_sequence_dict() -> {0: [1, 2], 1: [2]}
+      >>> g.to_sequence_dict(sequence_constructor=tuple) -> {0: (1, 2), 1: (2,)}
+      >>> g.vs['name'] = ['apple', 'pear', 'peach']
+      >>> g.to_sequence_dict(use_vids=False)
+      {'apple': ['pear', 'peach'], 'pear': ['peach']}
+
+    @param use_vids: (bool) whether to label vertices in the output data
       structure by their ids or their vertex_name_attr attribute. If
       use_vids=False but vertices lack a vertex_name_attr attribute, an
       AttributeError is raised.
-    @param sequence_constructor (function): constructor for the data structure
+    @param sequence_constructor: (function) constructor for the data structure
```

Comment:
I'd remove this type annotation unless you can come up with something precise that we can put in the function header (probably something like `Callable[[], Sequence[int]]` but I'm not sure whether it would pass the scrutiny of `mypy` or `pylance`).

Reply:
I annotated it as a generic `callable` for now. Not as fancy, but will catch some fish. Can be improved later on, when we add type annotatiions everywhere... which we will do, perhaps?

Reply:
Type annotations cannot be added to functions that are implemented directly in C, so we would either need an interim layer in Python that simply forwards all arguments to the C implementation, or we can use something like https://github.com/igraph/python-igraph-ctypes instead of hand-written C code, and then we can use Stimulus to generate the typings as well.

Reply:
yes yes I was just joking since it's one of those things that are easier said than done :smile: